### PR TITLE
feat: Create `Radio` component (#562)

### PR DESCRIPTION
**Pull request by essj**
Issue [#562](https://github.com/binance-chain/ui-project-management/issues/562).

Had to fix / clean up some other components along the way.

Nov-02-2020 23-18-38

### CODE REVIEW

File: src/components/Card/component.tsx [hunk 1]
```diff
@@ -1,23 +1,32 @@
...
   children,
-  className,
+  htmlTag,
   'data-testid': testId,
   position = 'center',
   variant = 'default',
+  ...otherProps
 }: Props) => (
-  <Container className={className} data-testid={testId} position={position} variant={variant}>
+  <Container
```
Comment (cchungmin) [inline src/components/Card/component.tsx:23]: Looks like now  a spread of `otherProps` is introduced and `className` is abandoned only for the use in tests. Is this intended? (Looks like `className` is not applied though)

File: src/components/Card/component.tsx [hunk 2]
```diff
@@ -1,23 +1,32 @@
...
+    {...otherProps}
+    as={htmlTag as any}
```
Comment (cchungmin) [inline src/components/Card/component.tsx:25]: Just curious about `htmlTag as any`. It seems that we use this in some files. Is there any type we can use here? Not sure, maybe `HTMLElementTagNameMap`

File: src/components/Card/component.tsx (hunk 1 above)
Comment (essj) [reply src/components/Card/component.tsx:23]: Yes, I realised when making the stories for this component that other HTML attributes accepted as props were not being applied to the `Card` component (such as `style`). `className` is included in the spread.

File: src/components/Card/component.tsx (hunk 2 above)
Comment (essj) [reply src/components/Card/component.tsx:25]: At the moment we have:
```
export type HtmlTag = NonNullable<keyof React.ReactHTML>;
```

I'm not sure what the difference between this and `HTMLElementTagNameMap` is, I'll have to research or if you recommend we change it we can. Note that even if we use `HTMLElementTagNameMap`, it seems we still need the `as any` cast.

image

File: src/components/Card/component.tsx (hunk 1 above)
Comment (cchungmin) [reply src/components/Card/component.tsx:23]: Yea now any React native prop is acceptable. I think you could only use `className` and style your card in tests in `styled-component`. Generally I tend to pass only the props exactly needed down to the child, but let me know what you think.

File: src/components/Card/component.tsx (hunk 2 above)
Comment (cchungmin) [reply src/components/Card/component.tsx:25]: Yea we can do this later as a separate task. I am not also 100% sure. My comment here might be redundant :satisfied:

**Review by cchungmin — APPROVED** [review]

**Pull request merged**